packages/expect/src/index.ts

```diff
@@ -53,18 +52,24 @@ class JestAssertionError extends Error {
   matcherResult: any;
 }
 
-const isPromise = obj =>
+const isPromise = (obj: any): obj is PromiseLike<any> =>
```

Comment:
I've been thinking about this typing quite a bit. A comment on this would be appreciated

Reply:
Because its an arrow function, this generic would look weird. But it works, and that way you can forward the type of the promise. Also; if we're talking ecma promises, that should just read `obj is Promise`, but here is something.

```ts
const isPromise = <T extends any>(obj: any): obj is PromiseLike<T> => {
	return ...
}
```
> note, you have to extend _something_, or else the generic doesn't work 🤔